tests/Fixer/Comment/SingleLineCommentStyleFixerTest.php

```diff
@@ -0,0 +1,321 @@
+//',
+                '<?php
+/***
+ *
+ */',
+            ],
+            [
+                '<?php
+
+    // s',
+                '<?php
+
+    /***
+s    *
+     */',
+            ],
+
+            // Untouched cases
+            [
+                '<?php
+$a = 1; /* in code */ $b = 2;
+',
+            ],
+            [
+                '<?php
+    /*
+     * in code 2
+     */ $a = 1;
+',
+            ],
+            [
+                '<?php
+/***
+ *
+ */ $a = 1;',
+            ],
+            [
+                '<?php
+    /***
+s    *
+     */ $a = 1;',
+            ],
+            [
+                '<?php
+    /*
+     * first line
+     * second line
+     */',
+            ],
+            [
+                '<?php
+    /*
+     * first line
+     *
+     * second line
+     */',
+            ],
+            [
+                '<?php
+    /*first line
+second line*/',
+            ],
+            [
+                '<?php /** inline doc comment */',
+            ],
+            [
+                '<?php
+    /**
+     * Doc comment
+     */',
+            ],
+            [
+                '<?php # test',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideHashCases
+     */
+    public function testHashCases($expected, $input = null)
+    {
+        $this->fixer->configure(['comment_types' => ['hash']]);
+        $this->doTest($expected, $input);
+    }
+
+    public function provideHashCases()
+    {
+        return [
+            [
+                '<h1>This is an <?php //echo 123;?> example</h1>',
+                '<h1>This is an <?php #echo 123;?> example</h1>',
+            ],
+            [
+                '<?php
+                    // test
+                ',
+                '<?php
+                    # test
+                ',
+            ],
+            [
+                '<?php
+                    // test1
+                    //test2
+                    // test3
+                    // test 4
+                ',
+                '<?php
+                    # test1
+                    #test2
+                    # test3
+                    # test 4
+                ',
+            ],
+
+            // Untouched cases
+            [
+                '<?php
+                    //#test
+                ',
+            ],
+            [
+                '<?php
+                    /*
+                        #test
+                    */
+                ',
+            ],
+            [
+                '<?php // a',
+                '<?php # a',
+            ],
+            [
+                '<?php /* start-end */',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideAllCases
+     */
+    public function testAllCases($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
```

Comment:
This should be tested in the code that handles allowed values + config default, shouldn't it?
Do you have records of a fixer that misbehaved that way, since the time new config system was introduced?